snakemake_executor_plugin_aws_batch/batch_job_builder.py

```
@@ -0,0 +1,96 @@
+            job_def = self.batch_client.client.register_job_definition(
+                jobDefinitionName=job_definition_name,
+                type=BATCH_JOB_DEFINITION_TYPE.CONTAINER.value,
+                containerProperties=container_properties,
+                timeout=timeout,
+                tags=tags,
+                platformCapabilities=[BATCH_JOB_PLATFORM_CAPABILITIES.EC2.value],
+            )
+            self.created_job_defs.append(job_def)
+            return job_def, job_name
+        except Exception as e:
+            raise WorkflowError(e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)